.travis.yml

```diff
@@ -4,9 +4,18 @@ os: linux
 
 sudo: required
 
-compiler: gcc
+compiler:
+  - gcc
+  - clang
+
+env:
+  - CONFIGURE_CMD=configure
+  - CONFIGURE_CMD=cmake
 
 script:
-  - ./configure
+  - mkdir -p build
+  - cd build
```

Comment:
Cd outside the build and then run configure

Reply:
so with configure, we can not make an out-of-source build?
why not.. and can we "fix" that?

Reply:
Basically the configuration file resides in the root directory of the project as a standard. Since you cd into build, it won't be there in the build folder and you are running ./configure which will make the shell to execute the configure file in ./ i.e, current folder(build).